### PR TITLE
Fix errors when trying to DELETE FROM AO/CO auxiliary tables

### PR DESCRIPTION
**Pull request by jimmyyih**
When the AO/CO auxiliary tables have data and a DELETE is called on them after setting allow_system_table_mods to on, the DELETE would fail due to assertion failure. The issue was that the rewrite handler and the executor did not know about AO/CO auxiliary tables so the junk filters were not being set up correctly like they would be if the AO/CO auxiliary tables were regular heap tables. To fix the issue, simply add logic to check for the AO/CO auxiliary table relkinds.

Issue was reported from gpupgrade team where this DELETE operation is done in the data migration scripts.

### CODE REVIEW

File: src/backend/executor/nodeModifyTable.c [hunk 1]
```diff
@@ -2422,7 +2423,8 @@ ExecModifyTable(PlanState *pstate)
 
 				relkind = resultRelInfo->ri_RelationDesc->rd_rel->relkind;
 				if (relkind == RELKIND_RELATION || relkind == RELKIND_MATVIEW ||
-					relkind == RELKIND_PARTITIONED_TABLE)
+					relkind == RELKIND_PARTITIONED_TABLE ||
+					IsAppendonlyMetadataRelkind(relkind))
```
Comment (jimmyyih) [inline src/backend/executor/nodeModifyTable.c:2427]: IMO, this `IsAppendonlyMetadataRelkind()` function from `catalog/aocatalog.h` should really be a macro in `pg_class.h` instead. Just something I wanted to note down to see if others agree or not.

File: src/backend/rewrite/rewriteHandler.c [hunk 2]
```diff
@@ -43,6 +43,8 @@
 #include "utils/lsyscache.h"
 #include "utils/rel.h"
 
+#include "catalog/aocatalog.h"
```
Comment (kalensk) [inline src/backend/rewrite/rewriteHandler.c:46]: nit: should this include be in alphabetical order?

File: src/test/regress/sql/uao_catalog_tables.sql [hunk 3]
```diff
@@ -127,3 +132,71 @@ update uao_table_check_hidden_tup_count_after_update set j = 'test21';
 select * from gp_toolkit.__gp_aovisimap_hidden_info('uao_table_check_hidden_tup_count_after_update'::regclass);
 vacuum full uao_table_check_hidden_tup_count_after_update;
 select * from gp_toolkit.__gp_aovisimap_hidden_info('uao_table_check_hidden_tup_count_after_update'::regclass);
+
+-- Verify that the AO auxiliary tables can be deleted from with allow_system_table_mods set
```
Comment (kalensk) [inline src/test/regress/sql/uao_catalog_tables.sql:136]: nit: remove "from" => "... can be deleted when allow_system_table_mods is set"

**Review by kalensk — APPROVED** [review]

File: src/backend/executor/nodeModifyTable.c (hunk 1 above)
Comment (bmdoil) [reply src/backend/executor/nodeModifyTable.c:2427]: I agree that it should be a macro in pg_class.h. 
Frontend code could also utilize it via pg_class_d.h [instead of re-implementing the check
](https://github.com/greenplum-db/gpdb/blob/master/src/bin/pg_upgrade/greenplum/aotable.c#L283-L287)

**Review by bmdoil — APPROVED** [review]

File: src/backend/rewrite/rewriteHandler.c (hunk 2 above)
Comment (jimmyyih) [reply src/backend/rewrite/rewriteHandler.c:46]: It's separated out from the above to make GPDB postgres merges easier.  This is common in the GPDB code.

File: src/test/regress/sql/uao_catalog_tables.sql (hunk 3 above)
Comment (jimmyyih) [reply src/test/regress/sql/uao_catalog_tables.sql:136]: "deleted from" is definitely the correct phrase here since we're deleting from the AO auxiliary tables and not deleting the tables themselves.  If it makes it easier, I could maybe reword this to "Verify that we can delete from the AO auxiliary tables when allow_system_table_mods is enabled"?

Comment (kalensk) [reply src/test/regress/sql/uao_catalog_tables.sql:136]: Your updated phrasing is much more clear to me than the original.
